ruby/ext/google/protobuf_c/map.c

```diff
@@ -167,9 +167,9 @@ void Map_free(void* _self) {
 
 VALUE Map_alloc(VALUE klass) {
   Map* self = ALLOC(Map);
+  VALUE ret = TypedData_Wrap_Struct(klass, &Map_type, self);
   memset(self, 0, sizeof(Map));
   self->value_type_class = Qnil;
-  VALUE ret = TypedData_Wrap_Struct(klass, &Map_type, self);
   return ret;
```

Comment:
Maybe just return the result of TypedData_Wrap_Struct directly, to avoid having an unclean RObject on the stack prematurely.